query/logicalplan/optimize_test.go

```diff
@@ -22,20 +22,21 @@ func TestOptimizePhysicalProjectionPushDown(t *testing.T) {
 		Build()
 
 	optimizer := &PhysicalProjectionPushDown{}
-	optimizer.Optimize(p)
+	optimizer.Optimize(tableProvider.schema, p)
 
 	require.Equal(t, &TableScan{
 		TableName:     "table1",
 		TableProvider: tableProvider,
 		// Only these columns are needed to compute the result. There can be
-		// duplicates because the statements just add the matchers for the
-		// columns they access. The optimizer could potentially deduplicate or
-		// use a more efficient datastructure in the future.
+		// duplicates because the statements of separate query plans just add
+		// the matchers for the columns they access. The optimizer could
+		// potentially deduplicate or use a more efficient datastructures
+		// in the future.
 		Projection: []ColumnMatcher{
 			StaticColumnMatcher{ColumnName: "stacktrace"},
 			StaticColumnMatcher{ColumnName: "stacktrace"},
 			StaticColumnMatcher{ColumnName: "value"},
-			StaticColumnMatcher{ColumnName: "labels.test"},
+			DynamicColumnMatcher{ColumnName: "labels.test"},
```

Comment:
This doesn't look right to me. A `StaticColumnMatcher` is correct here because in the materialized result we intend to match on the `labels.test` column. A `DynamicColumnMatcher` is intended to match all `labels.*` columns. Like in an aggregation where we aggregate by all unique label combinations.

Reply:
Had to dig into why this needed to be changed again.
We have a projection for the `labels` column in some queries and for the `ArrowSchema` to be returned correctly we want to match all columns in the arrow schema that start with `labels.`. So yes in some cases we need to propagate the column as DynamicColumn. 
This is needed for the `includedProjection` in `arrow.go`.
![Screenshot from 2022-06-09 15-24-10](https://user-images.githubusercontent.com/872251/172857825-5e0e576d-0cd3-4e19-b432-4d7ce3abfe52.png)
The column `label` needs to be dynamically matched, so all columns starting with `labels.` are returned. Otherwise, the graphs will look like one series as no different label sets are returned.
![Screenshot from 2022-06-09 15-26-00](https://user-images.githubusercontent.com/872251/172858171-1dceb20b-fd5c-45fb-9a4d-574fc724097d.png)

Reply:
Addition: Once querying a specific profile we know all columns exactly and the dynamic columns can then be treated as if they were StaticColumns since we know their full name like `labels.job` or `labels.instance`.

Reply:
That makes sense, but in that case `logicalplan.DynCol("labels")` should have been used which consequently correctly uses the DynamicColumnMatcher as opposed to the StaticColumnMatcher.

Reply:
Column names are sorted so the receiving side should not need to care about the concrete columns when querying, and instead merge them transparently.